src/WebJobs.Extensions.IoTHub/IoTHubConfigProvider.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Text;
+using Microsoft.Azure.Devices;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Extensions.IoTHub
+{
+    /// <summary>
+    /// Binding provider for IoTHub
+    /// </summary>
+    public class IoTHubConfigProvider : IExtensionConfigProvider, IDisposable
+    {
+        private readonly ConcurrentDictionary<string, ServiceClient> ClientCache = new ConcurrentDictionary<string, ServiceClient>();
+        private readonly static JsonSerializer _serializer = new JsonSerializer();
+        private bool _disposed = false;
+
+        /// <inheritdoc/>
+        public void Initialize(ExtensionConfigContext context)
```

Comment:
Does this need any configuration from host.json? 
You can now pull that via context.ApplyConfig(this, "iot");